bpo-40823: Use loadTestsFromTestCase() iso. makeSuite() in sqlite3 tests

erlend-aasland opened the pull request:
<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
bpo-NNNN: Summary of the changes made
```

Where: bpo-NNNN refers to the issue number in the https://bugs.python.org.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `master`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `master`.

-->


<!-- issue-number: [bpo-40823](https://bugs.python.org/issue40823) -->
https://bugs.python.org/issue40823
<!-- /issue-number -->

Commit:
Use loadTestsFromTestCase iso. makeSuite in Lib/sqlite3/test/

Ref. issue 5846: unittest.makeSuite is deprecated

Commit:
Add NEWS

berkerpeksag reviewed (changes requested):
I don't have a strong objection to this, but if we are going to drop `Check` prefix, I'd prefer using `test_*` instead of `test*`.

erlend-aasland commented on Lib/sqlite3/test/types.py:
```diff
@@ -313,19 +313,19 @@ def tearDown(self):
         self.assertEqual(self.cur.description[0][0], "1")
 
-    def CheckCursorDescriptionCTESMultipleColumns(self):
+    def test_cursor_description_ctes_multiple_columns(self):
```
```suggestion
    def test_cursor_description_cte_multiple_columns(self):
```

@berkerpeksag: ctes => cte?

berkerpeksag replied:
CTE sounds good to me.

berkerpeksag commented on Lib/sqlite3/test/userfunctions.py:
```diff
@@ -494,17 +494,18 @@ def authorizer_cb(action, arg1, arg2, dbname, source):
+        AggregateTests,
+        AuthorizerTests,
+        FunctionTests,
+
```
Nitpick: I'd drop the extra newline here.

berkerpeksag commented on Misc/NEWS.d/next/Tests/2020-05-30-13-39-22.bpo-40823.yB7K5w.rst:
```diff
@@ -0,0 +1,2 @@
+Use :meth:`unittest.TestLoader().loadTestsFromTestCase` iso.
```
Nitpick: I'd prefer using the long form of iso here.

berkerpeksag commented on Misc/NEWS.d/next/Tests/2020-05-30-13-39-22.bpo-40823.yB7K5w.rst:
```diff
@@ -0,0 +1,2 @@
+Use :meth:`unittest.TestLoader().loadTestsFromTestCase` iso.
+:meth:`unittest.makeSuite` in :mod:`sqlite3` tests.
```
Nitpick: Please add "Patch by Your Name."

Commit:
Fix naming

test_cursor_description_ctes_multiple_columns => test_cursor_description_cte_multiple_columns

Commit:
Sort userfunctions.py suites and remove extra newline

Commit:
Fix NEWS entry

Commit:
Revert encoding changes in commit fd4f6514

berkerpeksag reviewed (approved):
LGTM!
